src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -77,15 +81,37 @@ public void characters(char[] ch, int start, int length) throws SAXException {
                 super.characters(ch, start, length);
             }
         };
-        Source src = new SAXSource(xr, new InputSource(new StringReader(FileUtils.readFileToString(xmlFile))));
+        String str = FileUtils.readFileToString(xmlFile);
+        Source src = new SAXSource(xr, new InputSource(new StringReader(str)));
         final ByteArrayOutputStream result = new ByteArrayOutputStream();
         Result res = new StreamResult(result);
         Transformer transformer = TransformerFactory.newInstance().newTransformer();
         //omit xml declaration because of https://bugs.openjdk.java.net/browse/JDK-8035437
         transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
         transformer.setOutputProperty(OutputKeys.ENCODING, OUTPUT_ENCODING);
-        transformer.transform(src, res);
 
-        return result.toString("UTF-8");
+        try {
+            transformer.transform(src, res);
+            return result.toString("UTF-8");
+        } catch (TransformerException e) {
+            return findSecretFallback(str);
+        }
     }
+
+
+    private static String findSecretFallback(String xml) {
+        Matcher matcher = Pattern.compile(">\\{(.*)\\}<|>(.*)\\=<").matcher(xml);
+        while(matcher.find()) {
```

Comment:
Maybe makes sense to cache pattern and to avoid compiling it every time. Likely NIT since it's a fallback